fsspec/core.py

```diff
@@ -468,6 +468,35 @@ def open_local(url, mode="rb", **storage_options):
     return paths
 
 
+def cp(filea, fileb: OpenFile, callback=None, **open_kwargs):
```

Comment:
Note that the filesystem API `cp` has `recursive` and other parameters - it can do batch jobs. I had imagined the code here as a file-specific copy method of a filesystem which could be overidden in various specific implementations (like FTP!).

This code doesn't use the given callback.

Reply:
Thanks for this, I think it will be great to have this useful feature!

Sorry for the nitpick, but might be nice to use `src` and `dst` as argument names similar to the python [shutil.copyfile](https://docs.python.org/3/library/shutil.html#shutil.copyfile) function, also maybe mark them as positional only as well.

Reply:
Certainly fine with src/dst. I'm less sure there is a need to make them position-only.